lib/services.js

```diff
@@ -168,21 +168,25 @@ export default {
         // will reassign some of the above values with the ones
         // from the intiation of the multipart upload
         // (for instance, storage class)
-        Object.keys(metaHeaders).forEach(val => {
-            omVal[val] = metaHeaders[val];
-        });
-
-        if (overrideMetadata) {
-            Object.keys(overrideMetadata).forEach(val => {
-                omVal[val] = overrideMetadata[val];
+        if (!omVal.isDeleteMarker) {
```

Comment:
is this necessary? wouldn't there just be an empty array of metaHeaders if it's a delete marker?